File: src/main/java/leetcode/lists/hot100/LeetCode226.java
package leetcode.lists.hot100;

import common.TreeNode;

/**
 * @author wujingxinit@outlook.com
 * @date 2022/9/7 23:38
 */
public class LeetCode226 {

    public TreeNode invertTree(TreeNode root) {
        if (root == null) {
            return null;
        }
        TreeNode temp = root.left;
        root.left = root.right;
        root.right = temp;
        invertTree(root.left);
        invertTree(root.right);
        return root;
    }
}

File: src/main/go/leetcode/lists/hot100/LeetCode226.go
package hot100

func invertTree(root *TreeNode) *TreeNode {

	if root == nil {
		return nil
	}
	var invertTree func(*TreeNode)

	invertTree = func(node *TreeNode) {
		if node == nil {
			return
		}
		node.Left, node.Right = node.Right, node.Left
		invertTree(node.Left)
		invertTree(node.Right)
	}
	invertTree(root)
	return root
}
